application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java

```diff
@@ -150,15 +150,16 @@ private boolean isMemberInactive(Member member, Instant when) {
                     .and(HELP_CHANNEL_MESSAGES.SENT_AT.greaterThan(latestActiveMoment)))) == 0;
     }
 
-    private void pruneMemberFromRole(Member member, Role role, ForumChannel helpForum) {
+    private void pruneMemberFromRole(Member member, Role role, JDA jda) {
         Guild guild = member.getGuild();
+        TextChannel selectRolesChannel = getSelectRolesChannel(jda);
```

Comment:
ur executing this on each member who u prune, instead of finding the channel beforehand **once** and passing the channel down

Reply:
Im not sure if i understand, do you suggest to have a field which use `getRoleSelectChannel` method to get the channel and using that?

Reply:
i think i understand, you mean instead of passing down jda i should get the channel at the top and pass that instead.

Reply:
> i think i understand, you mean instead of passing down jda i should get the channel at the top and pass that instead.
yes